clippy_lints/src/vec.rs

```diff
@@ -241,7 +182,82 @@ impl UselessVec {
     }
 }
 
+#[derive(Copy, Clone)]
+pub(crate) enum SuggestedType {
+    /// Suggest using a slice `&[..]` / `&mut [..]`
+    SliceRef(Mutability),
+    /// Suggest using an array: `[..]`
+    Array,
+}
+
+impl SuggestedType {
+    fn desc(self) -> &'static str {
+        match self {
+            Self::SliceRef(_) => "a slice",
+            Self::Array => "an array",
+        }
+    }
+
+    fn snippet(
+        self,
+        cx: &LateContext<'_>,
+        args_span: Option<Span>,
+        len_span: Option<Span>,
+        app: &mut Applicability,
+    ) -> String {
+        let args = args_span
+            .map(|sp| snippet_with_applicability(cx, sp, "..", app))
+            .unwrap_or_default();
+        let maybe_len = len_span
+            .map(|sp| format!("; {}", snippet_with_applicability(cx, sp, "len", app)))
+            .unwrap_or_default();
```

Comment:
can this get an empty string, that would result in `; ` ?
I wonder if we should trim in this case to not suggest adding empty spaces in the suggestion, if that makes sense.

Reply:
Ahh, it might, if the span is empty in some rare occasions when macros involved (?).
Or, suggesting `; len` by default... which is not a correct suggestion anyway...

I'll make sure no suggestion is given in these scenarios, but I can't guarantee to come up with some test cases to test it lol :smile:

Reply:
Yeah, was just trying to think of "what could go wrong"